src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -96,14 +96,37 @@ export default class OrganizationDiscover extends React.Component {
       return;
     }
 
+    const newQuery = getQueryFromQueryString(search);
     // Clear data only if location.search is empty (reset has been called)
     if (!search && !params.savedQueryId) {
-      const newQuery = getQueryFromQueryString(search);
       queryBuilder.reset(newQuery);
       resultManager.reset();
       this.setState({
         data: resultManager.getAll(),
       });
+    } else if (search) {
+      // This indicates navigation changes (e.g. back button on browser)
+      // We need to update our search store and probably runQuery
+      const {projects, range, start, end} = newQuery;
+      let hasChange = false;
+
+      if (projects) {
+        this.updateProjects(projects);
+        hasChange = true;
+      }
+
+      if (range || (end && start)) {
+        this.updateDateTime({
+          period: range || null,
+          start: start || null,
+          end: end || null,
+        });
+        hasChange = true;
+      }
+
+      if (hasChange) {
+        this.runQuery();
```

Comment:
I'm not sure if this was intended, but I think it's a better experience when navigating back/forward that the updated query gets run. There's not a lot of test coverage with this though (i.e. I found bugs when switching to saved query view).

Reply:
It was. If we change this we now we should turn it off for the empty query state (no query params) so you don't get stuck in a loop

Reply:
Do you agree it's a better experience? If not we can take it out.

This is nested under:
```
     } else if (search) {
```
So it should only run if projects and a datetime period are present.

Reply:
I played around with it and it felt pretty nice, but I'm not really opinionated about it since saved searches autorun anyway. Ah yeah, it seems the case where there's no search or savedsearchid probably needs an explicit call to reset(), but I don't think that's relevant to this PR anyway, and we can tackle this optimization separately